sympy/stats/matrix_distributions.py

```diff
@@ -66,40 +66,46 @@ def rv(symbol, cls, args):
 
 class SampleMatrixScipy:
     """Returns the sample from scipy of the given distribution"""
-    def __new__(cls, dist, size):
-        return cls._sample_scipy(dist, size)
+    def __new__(cls, dist, size, seed=None):
+        return cls._sample_scipy(dist, size, seed)
 
     @classmethod
-    def _sample_scipy(cls, dist, size):
+    def _sample_scipy(cls, dist, size, seed):
         """Sample from SciPy."""
 
         from scipy import stats as scipy_stats
+        import numpy
         scipy_rv_map = {
-            'WishartDistribution': lambda dist, size: scipy_stats.wishart.rvs(
+            'WishartDistribution': lambda dist, size, rand_state: scipy_stats.wishart.rvs(
                 df=int(dist.n), scale=matrix2numpy(dist.scale_matrix, float), size=size),
-            'MatrixNormalDistribution': lambda dist, size: scipy_stats.matrix_normal.rvs(
+            'MatrixNormalDistribution': lambda dist, size, rand_state: scipy_stats.matrix_normal.rvs(
                 mean=matrix2numpy(dist.location_matrix, float),
                 rowcov=matrix2numpy(dist.scale_matrix_1, float),
-                colcov=matrix2numpy(dist.scale_matrix_2, float), size=size)
+                colcov=matrix2numpy(dist.scale_matrix_2, float), size=size, random_state=rand_state)
         }
 
         dist_list = scipy_rv_map.keys()
 
         if dist.__class__.__name__ not in dist_list:
             return None
 
-        return scipy_rv_map[dist.__class__.__name__](dist, size)
+        samples = []
+        rand_state = numpy.random.RandomState(seed=seed)
+        for _ in range(size[0]):
+            samp = scipy_rv_map[dist.__class__.__name__](dist, size[1], rand_state)
+            samples.append(samp)
+        return samples
```

Comment:
Does this fix something related to seeding or is it another issue?

Reply:
This is for sampling of matrices from matrix distributions. It is related to this issue and should be there in this PR.

Reply:
If this was previously just
```
return numpy_rv_map[dist.__class__.__name__](dist, size)
```
then can it not just be
```
return numpy_rv_map[dist.__class__.__name__](dist, size, state)
```
Why is the change to looping over `size[0]` needed?